main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -989,7 +1065,9 @@ object GenExpression {
         // Adding source line number for debugging
         addSourceLine(visitor, loc)
         // We push the 'length' of the array on top of stack
-        compileInt(visitor, exps.length, isLong = false)
+        val arrLengthCst = Ast.Constant.Int32(exps.length)
+        val arrLengthExp = Expr.Cst(arrLengthCst, MonoType.Int32, loc.asSynthetic)
+        compileExpression(arrLengthExp, visitor, currentClass, lenv0, entryPoint)
```

Comment:
Hmm. I did not realize compileInt was used in places like here.

Reply:
Maybe we just introduce a simple `compileInt` helper function that handles these cases and the `Int32` case?

Reply:
Could we maybe have compileInt and compileLong? Just to clear things up a bit?

Reply:
Sure thing